backend/src/main/java/asegroup1/api/daos/landregistry/LandRegistryDaoImpl.java

```diff
@@ -61,4 +71,50 @@ public void delete(String id) {
                     return landRegistryData;
                 }).collect(Collectors.toList());
     }
+    
+	public int updateAveragePrice(HashMap<String, Long> averagePrices) throws IOException, UnirestException {
+		int updatedRecords = 0;
+		EntityManager em = getEntityManager();
+
+		for (Entry<String, Long> averagePrice : averagePrices.entrySet()) {
+			PostCodeCoordinates coordsToUpdate = em.find(PostCodeCoordinates.class, averagePrice.getKey());
+
+			if (!coordsToUpdate.getAverageprice().equals(averagePrice.getValue())) {
+
+				// update local values
+				em.getTransaction().begin();
+				coordsToUpdate.setAverageprice(averagePrice.getValue());
+				em.merge(coordsToUpdate);
+
+				// write update to database
+				em.getTransaction().commit();
+				updatedRecords++;
+			}
+		}
+
+		em.close();
+
+		return updatedRecords;
+    }
+
+	@SuppressWarnings("unchecked")
+	public HashMap<String, List<String>> getAllPostcodes(String regex, boolean restrictToUnset, int groupCharSize) {
+		List<String> postcodes = (List<String>) getEntityManager()
+				.createNativeQuery("SELECT postcode FROM " + TABLE_NAME + "\n" + "WHERE postcode LIKE :outcode" + (restrictToUnset ? " AND averageprice = 0" : ""))
+				.setParameter("outcode", regex + "%").getResultList().stream().map(r -> String.valueOf(r)).collect(Collectors.toList());
```

Comment:
Can be cleaned up to `.getResultList().stream().map(String::valueOf).collect(Collectors.toList());`